libsplinter/src/admin/store/diesel/operations/list_circuits.rs

```diff
@@ -0,0 +1,263 @@
+            .collect::<Vec<String>>();
+        // Collects the members included in the list of `CircuitPredicates`
+        let members: Vec<String> = predicates
+            .iter()
+            .filter_map(|pred| match pred {
+                CircuitPredicate::MembersInclude(members) => Some(members.to_vec()),
+                _ => None,
+            })
+            .flatten()
+            .collect();
+
+        self.conn
+            .transaction::<Box<dyn ExactSizeIterator<Item = Circuit>>, _, _>(|| {
+                // Collects circuits which match the circuit predicates
+                let circuits: HashMap<String, CircuitModel> = circuit::table
+                    // Filters based on the circuit's management type
+                    .filter(circuit::circuit_management_type.eq_any(management_types))
+                    // Circuits are filtered by where there doesn't exist any `circuit_member` entries that
+                    // have a matching circuit_id value and have a node_id field that does not equal
+                    // any of the IDs collected from the `CircuitPredicates`.
+                    .filter(not(exists(
+                        // Selects all `circuit_member` entries where the `node_id` is not equal
+                        // to any of the members in the circuit predicates
+                        circuit_member::table.filter(
+                            circuit_member::circuit_id
+                                .eq(circuit::circuit_id)
+                                .and(circuit_member::node_id.ne_all(members)),
+                        ),
+                    )))
+                    .load::<CircuitModel>(self.conn)
+                    .map_err(|err| AdminServiceStoreError::QueryError {
+                        context: String::from("Unable to load Circuit information"),
+                        source: Box::new(err),
+                    })?
+                    // Once the `CircuitModels` have been collected, organize into a HashMap
+                    .into_iter()
+                    .map(|model| (model.circuit_id.to_string(), model))
+                    .collect();
+                // Store circuit IDs separately to make it easier to filter following queries
+                let circuit_ids: Vec<String> = circuits.keys().cloned().collect();
+
+                // Collect the `Circuit` members
+                let mut circuit_members: HashMap<String, Vec<String>> = HashMap::new();
+                for member in circuit_member::table
+                    .filter(circuit_member::circuit_id.eq_any(&circuit_ids))
+                    .load::<CircuitMemberModel>(self.conn)
+                    .map_err(|err| AdminServiceStoreError::QueryError {
+                        context: String::from("Unable to load Service information"),
+                        source: Box::new(err),
+                    })?
+                {
+                    if let Some(members) = circuit_members.get_mut(&member.node_id) {
```

Comment:
I think this should be using circuit_id not member.node_id, some for the insertions below.

Reply:
Yep, updated!